<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1061</width>
    <height>525</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Задание №4. Интерполяция</string>
  </property>
  <property name="windowIcon">
   <iconset resource="myresourse.qrc">
    <normaloff>:/img/img/icon.ico</normaloff>:/img/img/icon.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="icon">
        <iconset resource="myresourse.qrc">
         <normaloff>:/img/img/graphics_box.ico</normaloff>:/img/img/graphics_box.ico</iconset>
       </attribute>
       <attribute name="title">
        <string>Графики</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QCustomPlot" name="customPlot" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="icon">
        <iconset resource="myresourse.qrc">
         <normaloff>:/img/img/table.ico</normaloff>:/img/img/table.ico</iconset>
       </attribute>
       <attribute name="title">
        <string>Таблицы</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QTableWidget" name="tableWidget">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
            <weight>50</weight>
            <italic>true</italic>
            <bold>false</bold>
           </font>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>true</bool>
          </property>
          <property name="dragDropOverwriteMode">
           <bool>false</bool>
          </property>
          <property name="gridStyle">
           <enum>Qt::SolidLine</enum>
          </property>
          <property name="rowCount">
           <number>0</number>
          </property>
          <property name="columnCount">
           <number>4</number>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>240</number>
          </attribute>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>150</number>
          </attribute>
          <column/>
          <column/>
          <column/>
          <column/>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="icon">
        <iconset resource="myresourse.qrc">
         <normaloff>:/img/img/setting.ico</normaloff>:/img/img/setting.ico</iconset>
       </attribute>
       <attribute name="title">
        <string>Настройки</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <italic>true</italic>
              <bold>true</bold>
             </font>
            </property>
            <property name="title">
             <string>Настройки интерполяции</string>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <item row="2" column="0">
              <widget class="QGroupBox" name="groupBox">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <italic>true</italic>
                 <bold>true</bold>
                </font>
               </property>
               <property name="title">
                <string>Отрезок интерполирования</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_3">
                <item row="1" column="0">
                 <layout class="QVBoxLayout" name="verticalLayout_3">
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_3">
                    <item>
                     <widget class="QLabel" name="labelendSegment">
                      <property name="font">
                       <font>
                        <pointsize>10</pointsize>
                        <weight>75</weight>
                        <italic>true</italic>
                        <bold>true</bold>
                       </font>
                      </property>
                      <property name="text">
                       <string>Введите конец отрезка:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QDoubleSpinBox" name="doubleSpinBoxEndSegment">
                      <property name="enabled">
                       <bool>false</bool>
                      </property>
                      <property name="font">
                       <font>
                        <pointsize>8</pointsize>
                        <weight>50</weight>
                        <italic>false</italic>
                        <bold>false</bold>
                       </font>
                      </property>
                      <property name="decimals">
                       <number>4</number>
                      </property>
                      <property name="minimum">
                       <double>-100.000000000000000</double>
                      </property>
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                      <property name="singleStep">
                       <double>0.000100000000000</double>
                      </property>
                      <property name="value">
                       <double>1.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
                <item row="0" column="0">
                 <layout class="QVBoxLayout" name="verticalLayout_2">
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_2">
                    <item>
                     <widget class="QLabel" name="labelBeginSegment">
                      <property name="font">
                       <font>
                        <pointsize>10</pointsize>
                        <weight>75</weight>
                        <italic>true</italic>
                        <bold>true</bold>
                       </font>
                      </property>
                      <property name="text">
                       <string>Введите начало отрезка:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QDoubleSpinBox" name="doubleSpinBoxBeginSegment">
                      <property name="enabled">
                       <bool>false</bool>
                      </property>
                      <property name="font">
                       <font>
                        <pointsize>8</pointsize>
                        <weight>50</weight>
                        <italic>false</italic>
                        <bold>false</bold>
                       </font>
                      </property>
                      <property name="decimals">
                       <number>4</number>
                      </property>
                      <property name="minimum">
                       <double>-100.000000000000000</double>
                      </property>
                      <property name="maximum">
                       <double>100.000000000000000</double>
                      </property>
                      <property name="singleStep">
                       <double>0.000100000000000</double>
                      </property>
                      <property name="value">
                       <double>0.100000000000000</double>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSlider" name="horizontalSliderValueN">
               <property name="minimum">
                <number>4</number>
               </property>
               <property name="maximum">
                <number>14</number>
               </property>
               <property name="pageStep">
                <number>1</number>
               </property>
               <property name="sliderPosition">
                <number>4</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QLabel" name="labelValueN">
                 <property name="font">
                  <font>
                   <pointsize>10</pointsize>
                   <weight>75</weight>
                   <italic>true</italic>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Количество узлов интерполирования</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBoxValueN">
                 <property name="font">
                  <font>
                   <pointsize>8</pointsize>
                   <weight>50</weight>
                   <italic>false</italic>
                   <bold>false</bold>
                  </font>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:18pt; font-weight:600; font-style:italic; text-decoration: underline;&quot;&gt;Введите величину &amp;quot;n&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="statusTip">
                  <string/>
                 </property>
                 <property name="whatsThis">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:18pt; font-weight:600; font-style:italic; text-decoration: underline;&quot;&gt;Введите величину &amp;quot;n&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="minimum">
                  <number>4</number>
                 </property>
                 <property name="maximum">
                  <number>14</number>
                 </property>
                 <property name="value">
                  <number>4</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="3" column="0">
              <widget class="QGroupBox" name="groupBox_3">
               <property name="title">
                <string>Отрезок построения графика</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_5">
                  <item>
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>Начало построения по X:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QDoubleSpinBox" name="doubleSpinBoxBeginPlotsX">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                    <property name="minimum">
                     <double>-100.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>100.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.100000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_6">
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>Конец построения по X:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QDoubleSpinBox" name="doubleSpinBoxEndPlotsX">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                    <property name="minimum">
                     <double>-100.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>100.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.100000000000000</double>
                    </property>
                    <property name="value">
                     <double>2.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_7">
                  <item>
                   <widget class="QLabel" name="label_3">
                    <property name="text">
                     <string>Начало построения по Y:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QDoubleSpinBox" name="doubleSpinBoxBeginPlotsY">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                    <property name="minimum">
                     <double>-100.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>100.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.100000000000000</double>
                    </property>
                    <property name="value">
                     <double>-10.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_8">
                  <item>
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Конец построения по Y:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QDoubleSpinBox" name="doubleSpinBoxEndPlotsY">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                    <property name="minimum">
                     <double>-100.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>1000.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.100000000000000</double>
                    </property>
                    <property name="value">
                     <double>120.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
            <zorder>horizontalSliderValueN</zorder>
            <zorder>groupBox</zorder>
            <zorder>groupBox_3</zorder>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>503</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="labelInfo">
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <weight>75</weight>
        <italic>true</italic>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Информация</string>
      </property>
      <property name="textFormat">
       <enum>Qt::PlainText</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1061</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Справка</string>
    </property>
    <addaction name="action_aboutMe"/>
    <addaction name="action_Qt"/>
    <addaction name="action_QCustomPlot"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_QCustomPlot"/>
   <addaction name="action_Qt"/>
   <addaction name="action_aboutMe"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_aboutMe">
   <property name="icon">
    <iconset resource="myresourse.qrc">
     <normaloff>:/img/img/me.ico</normaloff>:/img/img/me.ico</iconset>
   </property>
   <property name="text">
    <string>Обо мне</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="action_Qt">
   <property name="icon">
    <iconset resource="myresourse.qrc">
     <normaloff>:/img/img/qt.ico</normaloff>:/img/img/qt.ico</iconset>
   </property>
   <property name="text">
    <string>Об Qt</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_QCustomPlot">
   <property name="icon">
    <iconset resource="myresourse.qrc">
     <normaloff>:/img/img/qcustomplot.ico</normaloff>:/img/img/qcustomplot.ico</iconset>
   </property>
   <property name="text">
    <string>Об QCustomPlot</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="myresourse.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>spinBoxValueN</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateValueN(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1048</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>533</x>
     <y>36</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxEndSegment</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateValueEndSegment(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1038</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxBeginSegment</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateValueBeginSegment(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1038</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>530</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderValueN</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBoxValueN</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>752</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>915</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxValueN</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSliderValueN</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>967</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>888</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateInterpolation()</slot>
  <slot>updateValueN(int)</slot>
  <slot>updateValueBeginSegment(double)</slot>
  <slot>updateValueEndSegment(double)</slot>
 </slots>
</ui>
